EasyPost/Exceptions/API/_Base.cs

```diff
@@ -64,23 +64,17 @@ protected ApiError(string errorMessage, int? statusCode = null, string? errorTyp
         // great minds think alike: https://github.com/stripe/stripe-dotnet/blob/6b9513d3b938d265c7607db919ad2c536ab578c3/src/Stripe.net/Infrastructure/Public/StripeClient.cs#L171
 
         /// <summary>
-        ///     Parse a errored RestResponse response object and return an instance of the appropriate exception class.
+        ///     Parse a errored HttpResponseMessage response object and return an instance of the appropriate exception class.
         ///     Do not pass a non-error response to this method.
         /// </summary>
-        /// <param name="response">RestResponse response to parse.</param>
-        /// <raises>EasyPostError if an unplanned response code is found (i.e. user passed in a non-error RestResponse response object with a 2xx status code).</raises>
+        /// <param name="response">HttpResponseMessage response to parse.</param>
+        /// <raises>EasyPostError if an unplanned response code is found (i.e. user passed in a non-error HttpResponseMessage response object with a 2xx status code).</raises>
         /// <returns>An instance of an HttpError-inherited exception.</returns>
-        internal static ApiError FromErrorResponse(RestResponse response)
+        internal static async Task<ApiError> FromErrorResponse(HttpResponseMessage response)
         {
             // NOTE: This method anticipates that the status code will be a non-2xx code.
             // Do not use this method to parse a successful response.
 
-            // short-circuit if the request timed out
```

Comment:
question: where is this now captured since this is getting reworked? Are we certain we still raise a timeout error?

Reply:
Yes, the unit test for it still passes. It's captured in the `ExecuteRequest` function in the client upstream and earlier